packages/inspire/VDOMView.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+import { vRefFromURI } from "~/raem/ValaaReference";
+
+import Cog from "~/engine/Cog";
+
+import { getGlobal } from "~/tools";
+
+/**
+ * This class is the view entry point
+ */
+export default class VDOMView extends Cog {
+  async initializeVDOM ({ name, size, rootLensURI }: Object) {
+    try {
+      if (!rootLensURI) {
+        throw new Error(`No options.rootLensURI found for view ${name}`);
+      }
+      // Load project
+      const lensRef = vRefFromURI(rootLensURI);
+      this._rootConnection = await this.engine.prophet.acquirePartitionConnection(
+          lensRef.partitionURI());
+      this._vUIRoot = await this.engine.getVrapper(
+          lensRef.rawId() || this._rootConnection.partitionRawId());
+      this.warnEvent(`initialize(): partition '${this._vUIRoot.get("name")}' UI root set:`,
```

Comment:
Ville is asking for if we can specify the lens property in the rootLensURI. 
In fact we will be able to once the ResourceURI spec is complete. The lens property name extraction probably happens here via some `this._rootLensName = getLensNameFromURI(rootLensURI)` - the _rootLensName needs to be then passed down to the ReactRoot which should forward it via appropriate props onwards.